src/carma-1-tenth/run_c1t_metric_analysis.py

```diff
@@ -0,0 +1,108 @@
+        self.assertTrue(np.all(crosstrack_errors < 0.4))
+
+    def test_C1T_05(self):
+        _, crosstrack_errors = plot_crosstrack_error(self.bag_dir, "/route_graph", show_plots=False)
+        self.assertTrue(np.all(crosstrack_errors < 0.2))
+
+    def test_C1T_06(self):
+        velocities, target_velocities = plot_vehicle_speed(self.bag_dir, show_plots=False)
+        max_target_speed = np.max(target_velocities)
+        speeds_on_straights = velocities[target_velocities == max_target_speed]
+        self.assertTrue(np.all(np.abs(max_target_speed - speeds_on_straights) < 0.2))
+
+    def test_C1T_07(self):
+        velocities, target_velocities = plot_vehicle_speed(self.bag_dir, show_plots=False)
+        max_target_speed = np.max(target_velocities)
+        # Determine which slowdowns are due to turns and which slowdowns are due to entering/exiting goals
+        stop_detected = True  # Vehicle will originally be stopped
+        buffer = 0  # Buffer used to count previous velocities that may or may not be due to entering/exiting a goal
+        wait_for_vehicle_to_accelerate = 0
+        relevant_velocity_idxs = []
+        for velocity in target_velocities:
+            if velocity == max_target_speed:
+                if wait_for_vehicle_to_accelerate > 5:
+                    relevant_velocity_idxs += (buffer + 1) * [True]
+                else:
+                    relevant_velocity_idxs += (buffer + 1) * [False]
+                wait_for_vehicle_to_accelerate += 1
+                buffer = 0
+                stop_detected = False
+            elif stop_detected:
+                relevant_velocity_idxs.append(False)    
+            elif velocity == 0.0:
+                relevant_velocity_idxs += (buffer + 1) * [False]
+                buffer = 0
+                stop_detected = True
+                wait_for_vehicle_to_accelerate = 0
+            else:
+                buffer += 1
+        speeds_on_turns = velocities[relevant_velocity_idxs]
+        self.assertTrue(np.all(speeds_on_turns > 0.5 * max_target_speed))
+
+    def test_C1T_08(self):
+        arrival_distances = check_distance_to_arrival(self.bag_dir)
+        self.assertTrue(np.all(arrival_distances < 0.2))
+
+    def test_C1T_09(self):
+        self.assertTrue(check_message_published(self.bag_dir, "/outgoing_mobility_operation"))
+
+    def test_C1T_10(self):
+        time_between_messages = check_message_timing(self.bag_dir, "/outgoing_mobility_operation", "/incoming_mobility_operation")
+        self.assertTrue(np.all(time_between_messages < 1.5))
+
+    def test_C1T_12(self):
+        self.assertTrue(check_port_drayage_ack(self.bag_dir, "PICKUP"))
+        self.assertTrue(check_port_drayage_ack(self.bag_dir, "DROPOFF"))
+        pass
+
+    def test_C1T_13(self):
+        self.assertTrue(check_port_drayage_ack(self.bag_dir, "HOLDING_AREA"))
+        pass
+    
+
+if __name__ == "__main__":
```

Comment:
I would also advocate that tests have a very concise summary of what they are in the terminal output. More brief than the comments but enough to generally know what is being tested at a glance

Reply:
Print statements added for each test case that fails